runtime/lua/vim/ui.lua

```diff
@@ -89,13 +89,13 @@ function M.input(opts, on_confirm)
     on_confirm = { on_confirm, 'function', false },
   })
 
-  local secret = opts['secret']
-  opts['secret'] = nil
-
   opts = (opts and not vim.tbl_isempty(opts)) and opts or vim.empty_dict()
 
+  local secret = opts.secret
+  opts.secret = nil
```

Comment:
Why is `opts` modified here?

Reply:
Because it's passed as options to `input()`

Reply:
Maybe I am misunderstanding something, but why does this matter? `vim.fn.input` should not care about a `secret` key, or does it? Isn't this a source of confusion if a user does this:
```lua
local opts = { secret = true }
vim.ui.input(opts)
-- No secret anymore, why?
vim.ui.input(opts)
```

Reply:
that's a good point.

you're saying that we can just pass the `secret` key in to `vim.fn.input` without removing it?

Reply:
Yes! In this case, the `secret` key will simply never be accessed.